core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -958,7 +958,12 @@ class AutoTuner(
         .getOrElse(autoTunerConfigsProvider.MAX_PARTITION_BYTES)
     val recommended =
       if (isCalculationEnabled("spark.sql.files.maxPartitionBytes")) {
-        calculateMaxPartitionBytes(maxPartitionProp)
+        // Ensure that the calculated value is within bounds
+        val lowerBound = autoTunerConfigsProvider.MAX_PARTITION_BYTES_LOWER_BOUND_MB
+        val upperBound = autoTunerConfigsProvider.MAX_PARTITION_BYTES_UPPER_BOUND_MB
+        calculateMaxPartitionBytesInMB(maxPartitionProp)
```

Comment:
Added bounds checking for max partition bytes. This ensures that the max partition bytes is not reduced to value smaller than parquet row group size.

Ref: https://parquet.apache.org/docs/file-format/configurations

cc: @kuhushukla

Reply:
Should we apply the lower-upper bound only when Parquet is used in the App?

Reply:
Thank you @revans2. The above reasoning makes sense. Let's say AutoTuner recommends a low value (say 16 MB). Now due to this there are tasks with no inputs. Now for the next run, AutoTuner should recommend an increased value. Since the lower bound logic is beyond the scope for this issue, I will file a separate issue for setting this.